src/ripple/app/tx/TransactionEngine.cpp

```diff
@@ -241,8 +241,17 @@ TransactionEngine::applyTransaction (
                 throw std::runtime_error ("Duplicate transaction applied to closed ledger");
             }
 
-            // Charge whatever fee they specified.
-            mLedger->destroyCoins (getNValue (txn.getTransactionFee ()));
+            // Charge whatever fee they specified. We break the encapsulation of
+            // STAmount here and use "special knowledge" - namely that a native
+            // amount is stored fully in the mantissa:
+            auto const fee = txn.getTransactionFee ();
```

Comment:
That also sounds like a good choice.  But will `destroyCoins` then need to break `STAmount`'s encapsulation?  Like I said, your call.